Pawns/DeusEx/Classes/Merchant.uc

```diff
@@ -29,6 +32,55 @@ function bool FilterDamageType(Pawn instigatedBy, Vector hitLocation,
     return Super.FilterDamageType(instigatedBy, hitLocation, offset, damageType);
 }
 
+function EnterConversationState(bool bFirstPerson, optional bool bAvoidState)
+{
+    local DXRando dxr;
+    local DXRHints hints;
+    local Conversation con;
+    local ConEvent ce;
+    local ConEventSpeech ces;
+    local int newHint;
+    local string hint, details;
+
+    foreach AllActors(class'DXRando', dxr) {
+        hints = DXRHints(dxr.FindModule(class'DXRHints'));
+        break;
+    }
+    foreach AllObjects(class'Conversation', con) {
+        if (con.description == "Merchant") {
+            break;
+        }
+    }
+
+    for (ce = con.eventList; ce != None; ce = ce.nextEvent) {
+        if (ConEventSpeech(ce) != None) {
+            ConEventSpeech(ce).conSpeech.speech = greetings[Rand(ArrayCount(greetings))];
+            break;
+        }
+    }
+
+    do {
+        newHint = hints.GetHint();
+        hint = hints.hints[newHint];
+        details = hints.details[newHint];
+    } until (
+        newHint != lastHint &&
+        hint != "Viewers, you could've prevented this with Crowd Control." &&
+        hint != "Don't forget you (the viewer!) can" &&
+        details != "We just shared your death publicly, go retweet it!" &&
+        !(MultiSkins[0] == Texture'DeusExCharacters.Skins.ChefTex0' && hint == "If you need a Hazmat suit")
```

Comment:
so this hint is not allowed to be shown, but the one about killing Le Merchant is? lol

`AddHint("You can kill Le Merchant and loot him", "if you don't have enough money.");`

also I feel like if this grows too much it might be cleaner to turn this into an infinite while loop, with continues and breaks

Reply:
instead of checking for the chef's hat texture, you could check for `CarcassType==Class'LeMerchantCarcass'

Reply:
I thought allowing the one about killing him is funny in a way allowing the one about him selling you a hazmat isn't.

Reply:
yea we can leave that lol

Reply:
"also I feel like if this grows too much it might be cleaner to turn this into an infinite while loop, with continues and breaks"

That was actually my original implementation. But `while(true)` rubs some people the wrong way and this seems reasonable for now.